src/NzbDrone.Core/Indexers/HDBits/HdBitsParser.cs

```diff
@@ -0,0 +1,104 @@
+using NzbDrone.Core.Parser.Model;
+using System.Net;
+using NzbDrone.Core.Indexers.Exceptions;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System.Web;
+using System.Collections.Specialized;
+
+namespace NzbDrone.Core.Indexers.HDBits
+{
+    public class HdBitsParser : IParseIndexerResponse
+    {
+        private readonly string _apiKey;
+
+        public HdBitsParser(string apiKey)
+        {
+            _apiKey = apiKey;
+        }
+
+        public IList<ReleaseInfo> ParseResponse(IndexerResponse indexerResponse)
+        {
+            var torrentInfos = new List<ReleaseInfo>();
+
+            if (indexerResponse.HttpResponse.StatusCode != HttpStatusCode.OK)
+            {
+                throw new IndexerException(
+                    indexerResponse,
+                    "Unexpected response status {0} code from API request",
+                    indexerResponse.HttpResponse.StatusCode);
+            }
+
+            var jsonResponse = JsonConvert.DeserializeObject<HdBitsResponse>(indexerResponse.Content);
+
+            if (jsonResponse.Status != StatusCode.Success)
+            {
+                throw new IndexerException(
+                    indexerResponse,
+                    @"HDBits API request returned status code {0} with message ""{1}""",
+                    jsonResponse.Status,
+                    jsonResponse.Message ?? "");
+            }
+
+            var responseData = jsonResponse.Data as JArray;
+            if (responseData == null)
+            {
+                throw new IndexerException(
+                    indexerResponse,
+                    "Indexer API call response missing result data");
+            }
+
+            var queryResults = responseData.ToObject<TorrentQueryResponse[]>();
+
+            foreach (var result in queryResults)
+            {
+                var id = result.Id;
+                torrentInfos.Add(new TorrentInfo()
+                {
+                    Guid = string.Format("HDB-{0}", id),
+                    Title = result.Name,
+                    Size = result.Size,
+                    DownloadUrl = GetDownloadUrl(id),
+                    InfoUrl = GetInfoUrl(id),
+                    Seeders = result.Seeders,
+                    Peers = result.Leechers + result.Seeders,
+                    PublishDate = result.Added
+                });
+            }
+
+            return torrentInfos.ToArray();
+        }
+
+        private string GetDownloadUrl(long torrentId)
+        {
+            var args = new NameValueCollection(2);
+            args["id"] = torrentId.ToString();
+            args["passkey"] = _apiKey;
+
+            return BuildUrl("https://hdbits.org/download.php", args);
+        }
+
+        private string GetInfoUrl(long torrentId)
+        {
+            var args = new NameValueCollection(1);
+            args["id"] = torrentId.ToString();
+
+            return BuildUrl("https://hdbits.org/details.php", args);
```

Comment:
What about use the BaseUrl in the settings? (Appending /api in the RequestGenerator). Allows ppl to change the root url if needed.